programmers-coding-test/soongu/greedy/숫자 카드 게임.js

```diff
@@ -0,0 +1,12 @@
+
+const rl = require("readline-sync");
+
+let [ n, m ] = rl.question("행과 열을 입력해주세요.\n>> ").split('');
+const result = [];
+
+for (let i = 0; i < n; i++) {
+  let [ data ] = rl.question("데이터를 입력해주세요.\n>> ").split('');
+  result.push(Math.min(data));
+}
+result.sort((a, b) => a - b);
+console.log(result[n - 1]);
```

Comment:
push와 sort 알고리즘을 없애고도 해결할 수 있다. 데이터가 방대할 수록 시간 효율은 좋아진다.
```suggestion
let max = 0;
for (let i = 0; i < n; i++) {
  let [ data ] = rl.question("데이터를 입력해주세요.\n>> ").split('');
  let min = Math.min(data);
  max = max < min ? min : max; // 실제로는 max, min 과 같은 변수명은 피하는 것이 좋다. data도 마찬가지.
}
console.log(max);
```